lib/oli_web/live/sections/overview_view.ex

```diff
@@ -144,4 +150,67 @@ defmodule OliWeb.Sections.OverviewView do
 
     {:noreply, push_redirect(socket, to: Routes.delivery_path(socket, :index))}
   end
+
+  def handle_event("show_delete_modal", _params, socket) do
+    section_has_student_enrollments =
+      Sections.list_student_enrollments(socket.assigns.section.slug) != []
+
+    message =
+      if section_has_student_enrollments do
+        """
+          This section has student data and will be archived rather than deleted.
+          Are you sure you want to archive it? You will no longer have access to the data.
+        """
+      else
+        """
+          This action cannot be undone. Are you sure you want to delete this section?
+        """
+      end
+
+    action = if section_has_student_enrollments, do: "Archive", else: "Delete"
```

Comment:
NIT: We can wrap this two if of the same condition into a tuple or something. Like `{message, action} = if section_has_student_enrollments do ...`